e2e/key/keyprivate/keyprivate.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package keyprivate
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+var keyPath string
+var defaultKeyFile string
+
+// corruptKey will take a ASCII key (kpath) and change some chars in it (corrupt it).
+func corruptKey(t *testing.T, kpath string) {
+	input, err := ioutil.ReadFile(kpath)
+	if err != nil {
+		t.Fatalf("Unable to read file: %v", err)
+	}
+
+	lines := strings.Split(string(input), "\n")
+
+	for i, line := range lines {
+		if strings.Contains(line, "B") {
+			lines[i] = "P"
+		}
+	}
+	output := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(kpath, []byte(output), 0644)
+	if err != nil {
+		t.Fatalf("Unable to write to file: %v", err)
+	}
+}
+
+func testPrivateKey(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    []string
+		stdin   string
+		file    string
+		corrupt bool
+		succeed bool
+	}{
+		{
+			name:    "export_private",
+			args:    []string{"export", "--secret"},
+			stdin:   "0\n", // TODO: this will need to be '1' at some point in time -> issue #3199
```

Comment:
can you be more explicit as to what's the meaning of that 0 and why it's changing to a 1?

Reply:
Take a look at this issue https://github.com/sylabs/singularity/issues/3199.

Reply:
I know I can look at the issue. The issue is too far away when I'm looking at code. It's good to document that there's an open issue related to this, but it's not enough.

In this specific case, something like "the first entry in the selection list is item 0; see issue #3199"